### ISSUE TITLE
'toaster' in repos.conf has name different from repository name

### ISSUE THREAD

**Issue by nielsd** [toaster/gentoo-overlay#2]
...just a small thing:

!!! Section 'toaster' in repos.conf has name different from repository name 'Tilos-Gentoo-Overlay' set inside repository

many thanks,
sorry for the noise...

Niels.

**Comment by toaster** [comment]
I don't have such a section in any `repos.conf` at all.
What did I miss?

**Comment by nielsd** [comment]
Hiho Tilo,


when adding the repo (by XML) to layman i get in /etc/portage/repos.conf/layman.conf
...
...
[toaster]
priority = 50
location = /var/lib/layman/toaster
layman-type = git
auto-sync = No
```
```

This leads to the described error - after changing the repo name/id to "Tilos-Gentoo-Overlay]" it works flawless:

...

[Tilos-Gentoo-Overlay]
priority = 50
location = /var/lib/layman/toaster
layman-type = git
auto-sync = No
...

I assume the problems is the "repo_name" "Tilos Gentoo Overlay" which correctly should be "toaster". Doing it vice versa should be OK too, but it seems that the safer convention is the use a single name/id for the repo.name, any overlay "ids", foder names etc.pp ("toaster").


Depending from your layman setup you may still have a single "repo.conf" file for any repos instead of the folder " /etc/portage/repos.conf/" which - for me - seems part of a newer layout within Gentoo (but i did not followed that changes in detail).

hope this helps in any way.
many thanks for your work.


Niels.

**Comment by toaster** [comment]
I have the `repos.conf` directory but no `layman.conf` in there.
However, I will change the repo name.

**Comment by toaster** [comment]
Hi Niels,

since I neither have the file nor the problem: could you please check if the issue is fixed by f0acf1d?

**Comment by nielsd** [comment]
Yes, t's fixed now.

It seems you still have repos.conf in asingle file. Gentoo "recommends" at different places to change to a folder with multiple config files - i.e.:

https://wiki.gentoo.org/wiki/Project:Portage/Sync#Portage_configuration
https://wiki.gentoo.org/wiki/Layman#Installation_of_version_.28.3C.3D2.0.0.29

while the "old" single file based config is still unterstood.

This seems why peoples still have different config structures here.

```
gongo ~ # layman -d toaster

 * Deleting selected overlay(s)...
 * Deleting directory "/var/lib/layman/toaster"
 * Successfully deleted overlay(s) toaster.

gongo ~ # layman -a toaster

 * Adding overlay...
 * Overlay "toaster" is not official. Continue installing? [y/n]: y
 * Running Git... # ( cd /var/lib/layman  && /usr/bin/git clone https://github.com/toaster/gentoo-overlay.git /var/lib/layman/toaster )
Klone nach '/var/lib/layman/toaster' ...
remote: Counting objects: 684, done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 684 (delta 0), reused 0 (delta 0), pack-reused 680
Empfange Objekte: 100% (684/684), 254.19 KiB | 0 bytes/s, Fertig.
Löse Unterschiede auf: 100% (237/237), Fertig.
 * Running Git... # ( cd /var/lib/layman/toaster  && /usr/bin/git config user.name "layman" )
 * Running Git... # ( cd /var/lib/layman/toaster  && /usr/bin/git config user.email "layman@localhost" )
!!! Section 'Tilos-Gentoo-Overlay' in repos.conf has name different from repository name 'toaster' set inside repository
 * Successfully added overlay(s) toaster.
```
(and no warning/error within emerge jobs anymore - ß)

many thanks again.
best regards,


Niels.
